app/javascript/packs/referral_charts.jsx

```diff
@@ -1,209 +1,93 @@
-    var canvas = document.getElementById( "chart-canvas" )
-    if ( canvas ) {
-      document.removeChild( canvas )
-    }
-    canvas = document.createElement( "canvas" )
-    canvas.setAttribute( "id", "chart-canvas" )
-    canvas.setAttribute( "width", "400" )
-    canvas.setAttribute( "height", "300" )
-    wrapper.appendChild( canvas )
-
-    Chart.defaults.global.defaultFontFamily = "Poppins"
-    Chart.scaleService.updateScaleDefaults( "logarithmic", {
-      ticks: {
-        callback: function ( ...args ) {
-          // new default function here
-          const value = Chart.Ticks.formatters.logarithmic.call( this, ...args )
-          if ( value.length ) {
-            var numericalValue = Number( value )
-            return numericalValue >= 1 ? numericalValue.toLocaleString() : ""
-          }
-          return value
-        }
-      }
-    } )
-
-    new Chart( canvas, {
-      type: "line",
-      data: {
-        labels: this.createLabels( data[ 0 ][ "ymd" ] ),
-        datasets: [
-          {
-            label: "Downloads",
-            data: data.map( x => x.retrievals ),
-            borderColor: "#F88469",
-            fill: false
-          },
-          {
-            label: "Installs",
-            data: data.map( x => x.first_runs ),
-            borderColor: "#66C3FC",
-            fill: false
-          },
-          {
-            label: "30-Day-Use",
-            data: data.map( x => x.finalized ),
-            borderColor: "#7B82E1",
-            fill: false
-          }
-        ]
-      },
-      options: {
-        tooltips: {
-          mode: "x"
-        },
-        elements: {
-          point: {
-            radius: 0,
-            hitRadius: 5,
-            hoverRadius: 5
-          }
-        },
-        title: {
-          fontSize: 18,
-          display: true,
-          text: title.toUpperCase()
-        },
-        scales: {
-          xAxes: [
-            {
-              type: "time",
-              distribution: "series",
-              ticks: {
-                autoSkip: true,
-                source: "labels"
-              },
-              time: {
-                unit: "day"
-              }
-            }
-          ],
-          yAxes: [
-            {
-              type: "logarithmic",
-              ticks: {
-                suggestedMax: suggestedMax
-              }
-            }
-          ]
-        },
-        legend: {
-          labels: {
-            usePointStyle: true
-          }
-        }
-      }
-    } )
-  }
-
-  async viewReferralCodeStats () {
-    const node = this.selectMenuRef.current
+  async viewReferralCodeStats() {
+    const node = this.selectMenuRef.current;
     var url = routes.publishers.promo_registrations.show.path.replace(
       "{id}",
-      document.getElementById( "publisher_id" ).value
-    )
-    url = url.replace( "{referral_code}", node.state.value )
-    const result = await fetch( url, {
+      document.getElementById("publisher_id").value
```

Comment:
Any way we could pass this in as a prop?

Reply:
Done! https://github.com/brave-intl/publishers/pull/1807/commits/ce6476cb973b1e5a281a9233593a58803e588ee9